JBBCode/Parser.php

```diff
@@ -97,7 +97,7 @@ public function addBBCode($tagName, $replace, $useOption = false, $parseContent
      */
     public function addCodeDefinition(CodeDefinition $definition)
     {
-        $this->bbcodes[] = $definition;
+        $this->bbcodes[$definition->getTagName()][$definition->usesOption()] = $definition;
```

Comment:
Looks great, but I believe this will throw a warning if you don't explicitly initialize the nested array.